main.cc

```diff
@@ -60,7 +73,17 @@ int main(int argc, char *argv[]) {
       }
       break;
     case smctemp::kOpReadCpuTemp:
-      std::cout << std::fixed << std::setprecision(1) << smc_temp.GetCpuTemp();
+      double temp = 0.0;
+      while (attempts > 0) {
+        temp = smc_temp.GetCpuTemp();
+        attempts--;
+        if (attempts > 0 && (temp == 0.0 || std::isnan(temp))) {
+          usleep(1000000);
+        } else {
+          break;
+        }
+      }
```

Comment:
It seems like we can simplify the logic. How about this?
```suggestion
      while (attempts > 0) {
        temp = smc_temp.GetCpuTemp();
        if (temp > 0.0) {
          break;
        } else {
          usleep(1000000);
          attempts--;
        }
      }
```